### PR TITLE
Feature/648 nats auth and authz

### PR DESCRIPTION
**Pull request by ognjenkurtic**
# Description

Adds docs and code changes to support NATS authorization

## Related Issue

#648 

## Motivation and Context

We had to make sure only Bpi Subjects subscribe to channels intended for their private communication. 

## How Has This Been Tested

Tested manually, by following the instructions from the readme and firiing creation of new bpi messages from Postman and via Nats.

## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)
- [] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/docs/nats/example-nats-server.conf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        { user: bpi_operator, password: "$2a$11$f5vKHKR2xuFhh/.oqjEuKuIJ2R3h0T.kCuukftqONF0gVR4zHYdW.", permissions: $bpi_operator_user },
+        # Bpi Subject 1
+        { user: 0x08872e27BC5d78F1FC4590803369492868A1FCCb, password: "sample_password1", permissions: { publish = "general", subscribe = "0x08872e27BC5d78F1FC4590803369492868A1FCCb"  } },
+        # Bpi Subject 2
+        { user: 0xF58e44db895C0fa1ca97d68E2F9123B187b789d4, password: "sample_password2", permissions: { publish = "general", subscribe = "0xF58e44db895C0fa1ca97d68E2F9123B187b789d4"  } },
+    ]
+}
+
+# This is for clustering multiple servers together.
+cluster {
```
Comment (skosito) [inline examples/bri-3/docs/nats/example-nats-server.conf:28]: do we need cluster at this point?

File: examples/bri-3/docs/nats/nats-configuration.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# NATS configuration
+
+We are relying on simple user authorization. Each BpiSubject is represented with a single user in Nats. Each Nats user is authorized for specific topics through the permissions map in nat server congif file. Currenly, we assume that the Bpi Admin will manage these users and their permssions. [Link](    https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro) to Nats docs.
```
Comment (skosito) [inline examples/bri-3/docs/nats/nats-configuration.md:3]: please clarify this is authz for nats only, to not confuse with general authz

also typos:
- congif
- currenly
- BPi
- `docker run -p 4222:4222 -d nats:alpine` -  (no need for - at the end, might be confusing or even make command invalid)
- docker cp <donwload_location>
- generaate 

something tells me there are more

File: examples/bri-3/README.md [hunk 3]
```diff
@@ -85,7 +86,7 @@ Run the following commands in the terminal:
 
 $ nats context add local --description "Localhost" # adds localhost server to nats cli
 
-$ nats pub general "<message>" # publishes new message on the general subject. Assumes two bpi subjects are added to the db with proper PKs.
+$ nats --user <username> --password <password> pub general "<message>" # publishes new message on the general subject. Assumes two bpi subjects are added to the db with proper PKs.
```
Comment (skosito) [inline examples/bri-3/README.md:89]: should this docs be grouped with messaging section above?

File: examples/bri-3/docs/nats/nats-configuration.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+NATS config that can be used in the development environment can be found in here *./example-nats-server.conf*
+
+[More](https://github.com/nats-io/nats-server/tree/main/server/configs) NATS sample configurations.
+
+## Authorization flow
+
+1) On server initialization, BPI operator will create a main BPI user with a password and add it to server configuration. Example can be found in *./example-nats-server.conf* file, bpi_operator user.
```
Comment (skosito) [inline examples/bri-3/docs/nats/nats-configuration.md:11]: how would this conf file be stored/backuped in real life scenario?

File: examples/bri-3/.env.sample [hunk 5]
```diff
@@ -10,3 +10,5 @@ GOERLI_SERVICE_DID="did:ethr:0x5:0x0aE6F58120E3e53E94FeB7BFeAA11C7CFF6F70F0"
 GEORLI_SERVICE_SIGNER_PRIVATE_KEY=""
 SERVICE_URL="bri-3"
 BPI_NATS_SERVER_URL="localhost:4222"
+BPI_NATS_SERVER_USER="bpi_operator"
+BPI_NATS_SERVER_PASS="liftboiliftboiliftboiliftboi1"
```
Comment (skosito) [inline examples/bri-3/.env.sample:14]: was `liftboiliftboiliftboiliftboi` taken?

**Review by skosito — APPROVED** [review]

File: examples/bri-3/docs/nats/example-nats-server.conf (hunk 1 above)
Comment (ognjenkurtic) [reply examples/bri-3/docs/nats/example-nats-server.conf:28]: Nope but it is default nats config so i decided to leave it there.

File: examples/bri-3/docs/nats/nats-configuration.md (hunk 4 above)
Comment (ognjenkurtic) [reply examples/bri-3/docs/nats/nats-configuration.md:11]: Up to the admin. You would probably use a docker volume with regular daily backups.

File: examples/bri-3/.env.sample (hunk 5 above)
Comment (ognjenkurtic) [reply examples/bri-3/.env.sample:14]: Even liftboi passwords are weak without a number.

File: examples/bri-3/README.md (hunk 3 above)
Comment (ognjenkurtic) [reply examples/bri-3/README.md]: Fixed

File: examples/bri-3/docs/nats/nats-configuration.md (hunk 2 above)
Comment (ognjenkurtic) [reply examples/bri-3/docs/nats/nats-configuration.md]: Fixed

**Review by skosito — APPROVED** [review]

**Review by Kasshern — APPROVED** [review]
Stellar.

**Pull request merged**